flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java

```diff
@@ -0,0 +1,641 @@
+	 *
+	 *     (v12) -+-> (v22)
+	 *
+	 *            ^
+	 *            |
+	 *       (pipelined)
+	 * </pre>
+	 * 2 regions. Each has 2 pipelined connected vertices.
+	 */
+	private JobGraph createStreamingJobGraph() {
+		final JobVertex v1 = new JobVertex("vertex1");
+		final JobVertex v2 = new JobVertex("vertex2");
+
+		v1.setParallelism(2);
+		v2.setParallelism(2);
+
+		v1.setInvokableClass(AbstractInvokable.class);
+		v2.setInvokableClass(AbstractInvokable.class);
+
+		v2.connectNewDataSetAsInput(v1, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED);
+
+		final JobGraph jobGraph = new JobGraph(v1, v2);
+		jobGraph.setScheduleMode(ScheduleMode.EAGER);
+
+		return jobGraph;
+	}
+
+	/**
+	 * Creating job graph as below (execution view).
+	 * It's a representative of batch job.
+	 * <pre>
+	 *     (v11) -+-> (v21)
+	 *            x
+	 *     (v12) -+-> (v22)
+	 *
+	 *            ^
+	 *            |
+	 *        (blocking)
+	 * </pre>
+	 * 4 regions. Each consists of one individual vertex.
+	 */
+	private JobGraph createBatchJobGraph() {
+		final JobVertex v1 = new JobVertex("vertex1");
+		final JobVertex v2 = new JobVertex("vertex2");
+
+		v1.setParallelism(2);
+		v2.setParallelism(2);
+
+		v1.setInvokableClass(AbstractInvokable.class);
+		v2.setInvokableClass(AbstractInvokable.class);
+
+		v2.connectNewDataSetAsInput(v1, DistributionPattern.ALL_TO_ALL, ResultPartitionType.BLOCKING);
+
+		final JobGraph jobGraph = new JobGraph(v1, v2);
+		jobGraph.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES);
+
+		return jobGraph;
+	}
+
+	private ExecutionGraph createExecutionGraph(final JobGraph jobGraph) throws Exception {
+		return createExecutionGraph(jobGraph, -1L);
+	}
+
+	private ExecutionGraph createExecutionGraph(final JobGraph jobGraph, final long checkpointId) throws Exception {
+		final RestartStrategy restartStrategy = new InfiniteDelayRestartStrategy(10);
+		return createExecutionGraph(jobGraph, restartStrategy, checkpointId);
+	}
+
+	private ExecutionGraph createExecutionGraph(
+		final JobGraph jobGraph,
+		final RestartStrategy restartStrategy) throws Exception {
+
+		return createExecutionGraph(jobGraph, restartStrategy, -1L);
+	}
+
+	private ExecutionGraph createExecutionGraph(
+		final JobGraph jobGraph,
+		final RestartStrategy restartStrategy,
+		final long checkpointId) throws Exception {
+
+		final SimpleSlotProvider slotProvider = new SimpleSlotProvider(jobGraph.getJobID(), 14);
+
+		final ExecutionGraph eg = new ExecutionGraph(
+			new DummyJobInformation(
+				jobGraph.getJobID(),
+				jobGraph.getName()),
+			TestingUtils.defaultExecutor(),
+			TestingUtils.defaultExecutor(),
+			AkkaUtils.getDefaultTimeout(),
+			restartStrategy,
+			TestAdaptedRestartPipelinedRegionStrategyNG::new,
+			slotProvider);
+		try {
+			eg.attachJobGraph(jobGraph.getVerticesSortedTopologicallyFromSources());
+		}
+		catch (JobException e) {
+			e.printStackTrace();
+			fail("Job failed with exception: " + e.getMessage());
```

Comment:
This is an anti-pattern. The `try-catch` should be removed:

- createExecutionGraph() already declares `throws Exception`
- Test should fail if an exception is thrown

Reply:
Will fix it.